src/ompl/geometric/planners/informedtrees/src/AITstar.cpp

```diff
@@ -203,9 +203,9 @@ namespace ompl
             graph_.clear();
             forwardQueue_.clear();
             reverseQueue_.clear();
-            solutionCost_ = objective_->infiniteCost();
-            approximateSolutionCost_ = objective_->infiniteCost();
-            approximateSolutionCostToGoal_ = objective_->infiniteCost();
+            solutionCost_ = ompl::base::Cost(std::numeric_limits<double>::signaling_NaN());
+            approximateSolutionCost_ = ompl::base::Cost(std::numeric_limits<double>::signaling_NaN());
+            approximateSolutionCostToGoal_ = ompl::base::Cost(std::numeric_limits<double>::signaling_NaN());
```

Comment:
Side-stepping the actually objective's `infiniteCost()` is no good. I would instead check if `objective_` exists and then reset these costs if it exists. These are reset in `setup()` as well. That is:
```cpp
if (objective_)
{
    solutionCost_ = objective_->infiniteCost();            
    approximateSolutionCost_ = objective_->infiniteCost();
    approximateSolutionCostToGoal_ = objective_->infiniteCost();
}
```

Reply:
I added the requested changes in [f742b9c](https://github.com/ompl/ompl/pull/996/commits/f742b9c43220db42416dae315e32a70fcc846708)

The reason I didn't go for this solution initially is that if the pointer does not exist, the cost won't be reset. However, as you said, those values are also set in the setup function, so I guess it is better to have a valid, outdated cost than a potentially invalid one.

Reply:
@zkingston I tested the suggested code, and there were no segmentation faults.